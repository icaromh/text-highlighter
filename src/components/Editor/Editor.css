.Editor {

}

.Editor__wrapper {
  height: calc(100vh - 100px);
  overflow: auto;
}

.Editor__content {
  font-size: 1.2rem;
  line-height: 1.3;
  text-align: justify;

  min-height: 70vh;
  max-width: 800px;
  padding: 3rem 2.2rem;
  margin: 0.5rem;

  outline: none;

  background-color: #fff;
  box-shadow: 0 0 0 0.75px #d1d1d1,0 0 3px 0.75px #ccc
}

@media (min-width: 768px) {
  .Editor__content {
    margin: 2rem auto;
  }
}
