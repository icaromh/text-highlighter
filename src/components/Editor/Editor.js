import React from 'react'
import PropTypes from 'prop-types'

const Editor = ({ value }) => (
  <div className="Editor">
    <div className="Editor__wrapper">
      <div
        className="Editor__content"
        spellCheck={false}
        dangerouslySetInnerHTML={{ __html: value }} // eslint-disable-line react/no-danger
        contentEditable
      />
    </div>
  </div>
)

Editor.propTypes = {
  value: PropTypes.string,
}

Editor.defaultProps = {
  value: '',
}


export default Editor
