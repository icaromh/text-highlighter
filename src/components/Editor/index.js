import Editor from './Editor'
import './Editor.css'

export default Editor
