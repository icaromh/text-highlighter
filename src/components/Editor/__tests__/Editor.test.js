import React from 'react'
import { shallow } from 'enzyme'

import Editor from '../Editor'

describe('#Editor', () => {
  test('expect to render', () => {
    const wrapper = shallow(<Editor />)
    expect(wrapper).toMatchSnapshot()
  })

  test('expect to render the dangerously html', () => {
    const value = 'Hey, dummy testing'
    const wrapper = shallow(<Editor value={value} />)

    expect(wrapper).toMatchSnapshot()
  })
})
