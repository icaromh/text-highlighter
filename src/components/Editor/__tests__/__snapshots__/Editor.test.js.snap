// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Editor expect to render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Editor
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="Editor__wrapper"
      >
        <div
          className="Editor__content"
          contentEditable={true}
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
          spellCheck={false}
        />
      </div>,
      "className": "Editor",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="Editor__content"
          contentEditable={true}
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
          spellCheck={false}
        />,
        "className": "Editor__wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "Editor__content",
          "contentEditable": true,
          "dangerouslySetInnerHTML": Object {
            "__html": "",
          },
          "spellCheck": false,
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="Editor__wrapper"
        >
          <div
            className="Editor__content"
            contentEditable={true}
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            spellCheck={false}
          />
        </div>,
        "className": "Editor",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="Editor__content"
            contentEditable={true}
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            spellCheck={false}
          />,
          "className": "Editor__wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "Editor__content",
            "contentEditable": true,
            "dangerouslySetInnerHTML": Object {
              "__html": "",
            },
            "spellCheck": false,
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`#Editor expect to render the dangerously html 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Editor
    value="Hey, dummy testing"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="Editor__wrapper"
      >
        <div
          className="Editor__content"
          contentEditable={true}
          dangerouslySetInnerHTML={
            Object {
              "__html": "Hey, dummy testing",
            }
          }
          spellCheck={false}
        />
      </div>,
      "className": "Editor",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="Editor__content"
          contentEditable={true}
          dangerouslySetInnerHTML={
            Object {
              "__html": "Hey, dummy testing",
            }
          }
          spellCheck={false}
        />,
        "className": "Editor__wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "Editor__content",
          "contentEditable": true,
          "dangerouslySetInnerHTML": Object {
            "__html": "Hey, dummy testing",
          },
          "spellCheck": false,
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="Editor__wrapper"
        >
          <div
            className="Editor__content"
            contentEditable={true}
            dangerouslySetInnerHTML={
              Object {
                "__html": "Hey, dummy testing",
              }
            }
            spellCheck={false}
          />
        </div>,
        "className": "Editor",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="Editor__content"
            contentEditable={true}
            dangerouslySetInnerHTML={
              Object {
                "__html": "Hey, dummy testing",
              }
            }
            spellCheck={false}
          />,
          "className": "Editor__wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "Editor__content",
            "contentEditable": true,
            "dangerouslySetInnerHTML": Object {
              "__html": "Hey, dummy testing",
            },
            "spellCheck": false,
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
